pkg/cdi/spec.go

```diff
@@ -35,6 +35,9 @@ var (
 		"0.2.0": {},
 		"0.3.0": {},
 	}
+
+	// Externally set CDI Spec validation function.
+	specValidator func(*cdi.Spec) error
```

Comment:
Not critical for this change, but we could use an interface instead.

Reply:
You mean something like:

```
type SpecValidator interface {
    Validate(*cdi.Spec) error
}

```
Sure, we can do that. I'm just curious what would be the benefit to it ? Having state for the validator does not require it. Using a member function or a closure (like in `pkg/cdi/validate`) does that. If the idea is to have a more formal type defined, then I'd rather have a typedef'ed function. Something like this

```
// SpecValidator is a function for...
type SpecValidator func(*cdi.Spec) error
...
// SetSpecValidator specifies a function for...
func SetSpecValidator(v SpecValidator) {
    specValidator = v
}
```

Reply:
It would allow us to assign more complex validators at a later stage without requiring modification to the API. I don't feel too strongly about it though, so feel free to leave it as is for the purpose of this PR.

Reply:
Okay. Then let's keep it this way for now. The only real user for it right now is anyway just the `pkg/cdi.SetSpecValidator(pkg/cdi/validate.With*Spec()`-boilerplate to enable extra, schema-based validation, so currently it's more of an internal interface which just happens to be split up the way it is to keep part of the external dependencies optional.